teuthology/provision/openstack.py

```diff
@@ -57,52 +57,91 @@ def init_user_data(self, os_type, os_version):
         log.debug("user_data file contents:")
         log.debug(user_data)
 
+    def _create_volume(self, volume_name, size):
+        """
+        Create a volume
+        """
+        volume_id = None
+        try:
+            r = OpenStack().run("volume show -f json " +
+                                volume_name)
+            volume_id = self.get_value(json.loads(r), 'id')
+        except subprocess.CalledProcessError as e:
+            if 'No volume with a name or ID' not in e.output:
+                raise e
+        if volume_id:
+            log.warn("Volume {} already exists with ID {}; using it"
+                     .format(volume_name, volume_id))
+            return volume_id
```

Comment:
Remove this line, we do not want to try and reuse the lame volume which is left from previous executions.

Reply:
hm, do you want to create a new volume (with the same name) in this case?

Reply:
I haven't checked, but I believe the volumes are removed by name in the teardown phase, and that removal would fail if there were two volumes with the same name.

Reply:
I believe it just removes all the volumes for a given name otherwise we need to fix that.

Reply:
According to _destroy_ method:
```
    def destroy(self):
        """
        Delete the name_or_id OpenStack instance.
        """
        if not self.exists():
            return True
        volumes = self.get_volumes()
        OpenStack().run("server set --name REMOVE-ME-" + self.name_or_id +
                        " " + self['id'])
        OpenStack().run("server delete --wait " + self['id'] +
                        " || true")
        for volume in volumes:
            OpenStack().run("volume set --name REMOVE-ME " + volume + " || true")
            OpenStack().run("volume delete " + volume + " || true")
        return True
```
and corresponding `get_volumes()` method:
```
    def get_volumes(self):
        """
        Return the uuid of the volumes attached to the name_or_id
        OpenStack instance.
        """
        volumes = self['os-extended-volumes:volumes_attached']
        return [volume['id'] for volume in volumes ]
```
There are ID's used.

Reply:
> hm, do you want to create a new volume (with the same name) in this case?

@smithfarm Actually we have no choice, because it is not possible to remove defective volume, if only not go to and manually rename it to something like _REMOVE-ME-bla-bla-bla_. From other side if someone else is using this volume, he will be disappointed. I **believe** that libcloud can easily handle the resources with the same name :-D